serpentTools/tests/test_microxs.py

```diff
@@ -0,0 +1,182 @@
+        univ = '0'
+        expectedValue = [1.60327E+00, 1.59830E+00, 1.59005E+00, 1.57731E+00,
+                         1.55586E+00, 1.54909E+00, 1.54820E+00, 1.53829E+00,
+                         1.55250E+00, 1.54046E+00, 1.54549E+00, 1.55092E+00,
+                         1.54549E+00, 1.53785E+00, 1.54761E+00, 1.54864E+00,
+                         1.53520E+00, 1.52489E+00, 1.49595E+00, 1.51506E+00,
+                         1.50824E+00, 1.50808E+00, 1.48335E+00, 1.45860E+00]
+
+        expectedUnc = [0.00351, 0.00086, 0.00052, 0.00025, 0.00023, 0.00020,
+                       0.00013, 0.00015, 0.00011, 0.00013, 0.00011, 0.00012,
+                       0.00015, 0.00016, 0.00012, 0.00019, 0.00018, 0.00026,
+                       0.00023, 0.00032, 0.00045, 0.00094, 0.00115, 0.00285]
+
+        numpy.testing.assert_equal(self.reader.fluxRatio[univ], expectedValue)
+        numpy.testing.assert_equal(self.reader.fluxUnc[univ], expectedUnc)
+
+    def test_fydata(self):
+        """Test fission yields for the reader."""
+        parent = 922350
+        energy = 2.53000E-08
+        fp = 531350
+        excpectedIndYield = numpy.array([2.92737E-02])
+        excpectedCumYield = numpy.array([6.28187E-02])
+        actualIndYield, actualCumYield = self.reader.getFY(parent, energy, fp)
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+    def test_getFY(self):
+        """Test the getFY method."""
+        # raises exception due to non-existing parent
+        with self.assertRaises(SerpentToolsException):
+            actualIndYield, actualCumYield = \
+                self.reader.getFY(500000, 2.53000E-08, 531350)
+
+        # closest energy check
+        actualIndYield, actualCumYield = self.reader.getFY(922350, 1E-05, 531350)
+        excpectedIndYield = numpy.array([2.92737E-02])
+        excpectedCumYield = numpy.array([6.28187E-02])
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+        # non-existing daughter
+        actualIndYield, actualCumYield = self.reader.getFY(922350, 1E-05, 9999)
+        excpectedIndYield = numpy.array([])
+        excpectedCumYield = numpy.array([])
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+    def test_getXS(self):
+        """Test the getXS method."""
+        # raises exception due to non-existing universes
+        with self.assertRaises(SerpentToolsException):
+            vals, unc = self.reader.getXS('2', 982490, 18, 0)
+        # raises exception due to wrong isotope formatting
+        with self.assertRaises(SerpentToolsException):
+            vals, unc = self.reader.getXS('0', 5000000, 18, 0)
+
+        # Test that values match
+        univ = '0'
```

Comment:
From what I've seen, there isn't a way to have universes with non-integer names in SERPENT. Rather than storing universes as strings of integers, can you instead store them as integers? This makes it easier to request universe data.

Reply:
In our case, the universe was '0' and could really be converted to int(0). 
However, the name of the universe can be 'FUEL' and in such a case, the data will be stored on:
XS_FUEL